13-TeV-examples/uproot_python/ttbar_analysis.ipynb

```diff
@@ -0,0 +1,758 @@
+    "                \"lep_type\", \"lep_E\", \"eventNumber\"\n",
+    "            ],\n",
+    "            library=\"ak\",  # Choose output type as awkward array\n",
+    "            entry_start=int(numevents * fraction * start_entry),\n",
+    "            entry_stop=numevents * fraction * (start_entry + 1)\n",
+    "            ):\n",
+    "\n",
+    "            nIn = len(data) # number of events in this batch\n",
+    "\n",
+    "            # Check triggerE, triggerM  conditions\n",
+    "            data = data[cut_trig(data.trigE,data.trigM)]\n",
+    "\n",
+    "            # Require exactly one lepton\n",
+    "            data = data[one_lep(data.lep_n)]\n",
+    "\n",
+    "            # cut on lepton lep_pt\n",
+    "            data = data[ak.flatten(cut_lep_pt(data.lep_pt))]\n",
+    "\n",
+    "            # Cut on missing transverse energy met_et\n",
+    "            data = data[cut_met_et(data.met_et)]\n",
+    "\n",
+    "            # Cut on W boson transverse mass\n",
+    "            W_mt_data = cut_W_mt(data.lep_pt, data.lep_phi, data.met_et, data.met_phi)\n",
+    "\n",
+    "            data['W_mass'] = ak.flatten(W_mt_data[0])\n",
+    "\n",
+    "            data = data[ak.flatten(W_mt_data[1])]\n",
+    "\n",
+    "            # Check Njet and Nbjet  conditions\n",
+    "            jets_data = cut_Njet_and_Nbjet(data.jet_pt,data.jet_MV2c10)\n",
+    "\n",
+    "            data['q_jets'] = jets_data[1]\n",
+    "\n",
+    "            data['b_jets'] = jets_data[2]\n",
+    "\n",
+    "            data['pt_jets'] = jets_data[3]\n",
+    "\n",
+    "            data = data[jets_data[0]]\n",
+    "\n",
+    "            if 'data' not in sample: # only do this for Monte Carlo simulation files\n",
+    "                # multiply all Monte Carlo weights and scale factors together to give total weight\n",
+    "                data['total_weight'] = calc_weight(data)\n",
+    "\n",
+    "            mtop_data = mtop(data.jet_pt ,data.jet_E ,data.jet_eta ,data.jet_phi ,data.jet_MV2c10 ,data.lep_pt, data.lep_eta ,data.lep_phi, data.lep_E, data.met_et, data.met_phi)\n",
+    "\n",
+    "            data['mtop'] = mtop_data[0]\n",
+    "\n",
+    "            data['mtop_1'] = mtop_data[1]\n",
+    "\n",
+    "            data['mtop_2'] = mtop_data[2]\n",
+    "\n",
+    "            nOut = len(data) # number of events passing cuts in this batch\n",
+    "            data_all.append(data) # append array from this batch\n",
+    "            elapsed = time.time() - start # time taken to process\n",
+    "            print(\"\\t\\t nIn: \"+str(nIn)+\",\\t nOut: \\t\"+str(nOut)+\"\\t in \"+str(round(elapsed,1))+\"s\") # events before and after\n",
+    "\n",
+    "    return ak.concatenate(data_all) # return array containing events passing all cuts\n",
+    "\n",
+    "\n",
+    "def get_data_from_files(start, local):\n",
```

Comment:
Here's a quick [set of suggestions on how to debug](https://chatgpt.com/share/822611d4-8332-4313-8e26-b9042e3d0583)